agent-control/src/sub_agent/on_host/supervisor.rs

```diff
@@ -160,9 +160,10 @@ impl NotStartedSupervisorOnHost {
             OnHostAgentVersionChecker::checked_new(self.agent_identity.agent_type_id.clone())?;
 
         Some(spawn_version_checker(
-            self.agent_identity.id.clone(),
+            self.agent_identity.id.as_str().to_string(),
```

Comment:
same here

Reply:
As this is for setting the thread name, I am passing a string directly instead of an agent id. For example, in the case of the CD version checker for AC, there's no `AgentID` involved! so it does not fit that the function accepts and `AgentID` but we can use a generic `String` as we do.